File: 0075-sort-colors/0075-sort-colors.py
class Solution(object):
    def sortColors(self, nums):
        n = len(nums)
        # Selection Sort 
        for i in range (n):
            min_idx = i
            for j in range (i,n):
                if nums[min_idx] > nums [j]:
                    min_idx = j

            # swap 
            nums[min_idx] , nums [i] = nums[i] , nums[min_idx]
        

File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int start = 0 , end = nums.length-1 , ptr = 0;
        while (ptr <= end){
            if (nums[ptr] == 0){
                int temp = nums[start];
                nums[start] = nums[ptr];
                nums[ptr] = temp;
                start++;
                ptr++;
            }
            else if (nums[ptr] == 1){
                ptr++;
            }
            else{
                int temp = nums[end];
                nums[end] = nums[ptr];
                nums[ptr] = temp;
                end --;
            }
        }
    }
}